admin_organization.go

```diff
@@ -0,0 +1,110 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ AdminOrganizations = (*adminOrganizations)(nil)
+
+// AdminOrganizations describes all of the admin organization related methods that the Terraform
+// Enterprise API supports. Note that admin settings are only available in Terraform Enterprise.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/admin/organizations.html
+type AdminOrganizations interface {
+	// Read attributes of an existing organization via admin API.
+	Read(ctx context.Context, organization string) (*AdminOrganization, error)
+
+	// Update attributes of an existing organization via admin API.
+	Update(ctx context.Context, organization string, options AdminOrganizationUpdateOptions) (*AdminOrganization, error)
+
+	// Delete an organization by its name via admin API
+	Delete(ctx context.Context, organization string) error
+}
+
+// adminOrganizations implements AdminOrganizations.
+type adminOrganizations struct {
+	client *Client
+}
+
+// AdminOrganization represents a Terraform Enterprise organization returned from the Admin API.
+type AdminOrganization struct {
+	Name string `jsonapi:"primary,organizations"`
+
+	AccessBetaTools                  bool   `jsonapi:"attr,access-beta-tools"`
+	ExternalID                       string `jsonapi:"attr,external-id"`
+	GlobalModuleSharing              bool   `jsonapi:"attr,global-module-sharing"`
```

Comment:
This is unreleased and should remain a part of #163; let's remove this and all references to it for now.

Reply:
I kept this here because it gets [returned via the API](https://www.terraform.io/docs/cloud/api/admin/organizations.html#sample-response-1) ([atlas](https://github.com/hashicorp/atlas/blob/master/app/serializers/api/v2/admin/organization_serializer.rb#L25)). So technically its available at the moment for TFE. 

```
go-tfe % curl --header "Authorization: Bearer <token>" --header "Content-Type: application/vnd.api+json" 
https://tfe-zone-28c39e24.ngrok.io/api/v2/admin/organizations/tst-64f9b3e9-68f1-fa2d-2e41-7f565f854b5a | jq .

{
  "data": {
    "id": "tst-64f9b3e9-68f1-fa2d-2e41-7f565f854b5a",
    "type": "organizations",
    "attributes": {
.....
      "sso-enabled": false,
      "global-module-sharing": false,
      "module-consuming-organizations-count": 0,
```

I can remove it if you'd like.